samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -162,8 +163,16 @@ public void onProcessorChange(List<String> processors) {
 
   void doOnProcessorChange(List<String> processors) {
     // if list of processors is empty - it means we are called from 'onBecomeLeader'
-    // TODO: Handle empty currentProcessorIds or duplicate processorIds in the list
+    // TODO: Handle empty currentProcessorIds.
     List<String> currentProcessorIds = getActualProcessorIds(processors);
+    List<String> uniqueProcessorIds = currentProcessorIds.stream()
```

Comment:
Makes sense. 

Initial idea was to generate jobmodel with unique processorIds(not swallow & return). Since it was a complicated change, it's punted for this release based upon discussion with boris. This was reminiscence of that.
 
Fixed.